qiskit/transpiler/preset_passmanagers/generate_preset_pass_manager.py

```diff
@@ -226,13 +228,20 @@ def generate_preset_pass_manager(
                 raise ValueError(
                     f"Gates with 3 or more qubits ({gate}) in `basis_gates` or `backend` are "
                     "incompatible with a custom `coupling_map`. To include 3-qubit or larger "
-                    " gates in the transpilation basis, use a custom `target` instance instead."
+                    " gates in the transpilation basis, provide a custom `target` instead."
                 )
 
     if target is None:
         if backend is not None and _no_loose_constraints:
             # If a backend is specified without loose constraints, use its target directly.
             target = backend.target
+        elif _adjust_dt:
+            # If a backend is specified with loose dt, use its target and adjust the dt value.
+            target = copy.deepcopy(backend.target)
+            target.dt = dt
+            # we must update the instruction properties to internally clear the value of
+            # _instruction_durations in the target so that the new dt is saved
+            target._instruction_durations = None
```

Comment:
From our offline conversation we should move the cache invalidation here to the setter for `target.dt`. Then we don't need to anything else here besides overriding the value of dt in a copy.

Reply:
After an internal discussion with @ElePT, this approach works. But it also worries me that if we try to do this directly from rust, the effect wouldn't be propagated correctly. Luckily, we are working on moving the remaining parts of the Target to be also controlled from Rust, so we'd have more control over this.